runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPluginConfiguration.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2018, Infinitay <https://github.com/Infinitay>
+ * Copyright (c) 2019, Sophie Buckley <https://github.com/sophiebuckley>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.kingdomofmiscellania;
+
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+@ConfigGroup("kingdomOfMiscellania")
+public interface KingdomPluginConfiguration extends Config
+{
+	@ConfigItem(
+		keyName = "maxCofferAmount",
+		name = "Max coffer amount",
+		description = "Set the max amount of coins that you would like in your coffer"
+	)
+	default int maxCofferAmount()
```

Comment:
This should use [range](https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/config/Range.java) as there is a maximum allowed amount of coins allowed in the coffer.

Reply:
Since there are 2 max coffer amounts (5m, then 7.5m after completing Royal Trouble), I was going to implement a check to see which of the max amounts I should use for the range, but I don't think I can do this as there isn't a value to use within [InventoryID](runelite-api/src/main/java/net/runelite/api/InventoryID.java) in order to carry out a check similar to the one for Kingdom of Miscellania. Should I just stick with the higher of the two max values, 7.5m?

Reply:
Each quest has a Varbit that stores its state, so just check the varbit related to the Royal Trouble Quest to see if you should use the higher amount. Look at how the Diary plugin uses them

Reply:
Thanks for the guidance! :) I'll try that.

Reply:
I neglected to read the usage case here. Since it's a range annotation, you'll have to use the larger 7.5m value. The Varbit stuff can still be used for how much the player needs to top up, since without Royal Trouble you only lose up to 50k a day

Reply:
Maybe the problem can be mitigated to changing the config to `How many days do you want to have stored in the coffer` which would allow you to do the calculations while aware of the varbit value. With/without would be 75k/50k a day i think. 

Also, please join our [Discord](https://discord.gg/mePCs8U), it makes holding conversations easier :)

Reply:
Even if the maximum is calculated within the plugin, the minimum (of 0) can be enforced using a `@Range` annotation.